src/DynamicMvvm.Uno/ViewModel/BatchingViewModelView.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+
+namespace Chinook.DynamicMvvm
+{
+	/// <summary>
+	/// This implementation of <see cref="IViewModelView"/> batches its operations to reduce the call count of <see cref="CoreDispatcher.RunAsync"/>.<br/>
+	/// When <see cref="ExecuteOnDispatcher"/> is invoked, the dispatcher operation is delayed by a small duration during which other <see cref="ExecuteOnDispatcher"/> invocations that may occur are accumulated.
+	/// After that short delay, all accumulated actions are executed within the same <see cref="CoreDispatcher.RunAsync"/>.
+	/// </summary>
+	public class BatchingViewModelView : IViewModelView
+	{
+		private readonly FrameworkElement _frameworkElement;
+		private readonly int _throttleDurationMs;
+		private readonly object _mutex = new object();
+		private readonly Queue<Request> _requests = new Queue<Request>(capacity: 128);
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="BatchingViewModelView"/> class.
+		/// </summary>
+		/// <param name="frameworkElement">The <see cref="FrameworkElement"/>.</param>
+		/// <param name="throttleDurationMs">The amount of time during which the batching accumulation occurs.</param>
+		public BatchingViewModelView(FrameworkElement frameworkElement, int throttleDurationMs = 10)
+		{
+			_frameworkElement = frameworkElement ?? throw new ArgumentNullException(nameof(frameworkElement));
+			_throttleDurationMs = throttleDurationMs;
```

Comment:
Are those ms or seconds ? Because the task delay line 67 takes 10, so sound like a 10s throttle by default ?

Reply:
nvm .. Task.Delay is ms by default :D